cluster/src/main/java/com/hartwig/pipeline/calling/structural/gridss/stage/Driver.java

```diff
@@ -12,52 +15,79 @@
+        sampleArguments.add(new SampleArgument("reference", referenceSampleName, referenceSamplePath));
+        return this;
     }
 
     @Override
     public List<BashCommand> bash(final OutputFile input, final OutputFile output) {
-        return Lists.newArrayList(new VersionedToolCommand(GRIDSS,
+        return List.of(new VersionedToolCommand(GRIDSS,
                 "gridss",
                 Versions.GRIDSS,
-                "--output",
-                output.path(),
-                "--assembly",
-                assemblyBamPath,
-                "--workingdir",
-                VmDirectories.OUTPUT,
-                "--reference",
-                resourceFiles.refGenomeFile(),
-                "--jar",
-                GridssJar.path(),
-                "--blacklist",
-                resourceFiles.gridssBlacklistBed(),
-                "--configuration",
-                resourceFiles.gridssPropertiesFile(),
-                "--labels",
-                referenceSample + "," + tumorSample,
-                "--jvmheap",
-                "31G",
-                "--externalaligner",
-                referenceBamPath,
-                tumorBamPath));
+                Stream.concat(Stream.of("--output",
+                        output.path(),
+                        "--assembly",
+                        assemblyBamPath,
+                        "--workingdir",
+                        VmDirectories.OUTPUT,
+                        "--reference",
+                        resourceFiles.refGenomeFile(),
+                        "--jar",
+                        GridssJar.path(),
+                        "--blacklist",
+                        resourceFiles.gridssBlacklistBed(),
+                        "--configuration",
+                        resourceFiles.gridssPropertiesFile(),
+                        "--labels",
+                        sampleArguments.stream().map(SampleArgument::getSampleName).collect(Collectors.joining(",")),
+                        "--jvmheap",
+                        "31G",
+                        "--externalaligner"), sampleArguments.stream().map(SampleArgument::getSamplePath)).collect(Collectors.toList())));
+    }
+
+    private static class SampleArgument implements Comparable<SampleArgument> {
+        private final String type;
+        private final String sampleName;
+        private final String samplePath;
+
+        private SampleArgument(final String type, final String sampleName, final String samplePath) {
+            this.type = type;
+            this.sampleName = sampleName;
+            this.samplePath = samplePath;
+        }
+
+        public String getSampleName() {
+            return sampleName;
+        }
+
+        public String getSamplePath() {
+            return samplePath;
+        }
+
+        @Override
+        public int compareTo(@NotNull final Driver.SampleArgument o) {
+            return this.type.compareTo(o.type);
```

Comment:
Am I understanding properly that this just maintains the order of reference first if it exists?

Reply:
yeah, i'll make it a bit more clear